### ISSUE TITLE
[Python] Incorrect code generated for integer properties with enum

### ISSUE THREAD

**Issue by zjarkov** [swagger-api/swagger-codegen#3667]
##### Description

Data type of enum items in Python generated code for property setter does not match to given yaml/json datatype (and/or expected type of property). It leads to permanent validation failures.

##### Swagger-codegen version

2.2.1

##### Swagger declaration file content or url

_yaml_
```yaml
baudrate:
  type: integer
  format: int32
  enum:
    - 1200
    - 2400
    - 4800
    - 9600
    - 19200
```

_json_
```json
"baudrate" : {
    "type" : "integer",
    "format" : "int32",
    "enum" : [1200, 2400, 4800, 9600, 19200]
}
```

##### Produced code

```python
@baudrate.setter
def baudrate(self, baudrate):
    """
    Sets the baudrate of this ModbusRtuSettings.

    :param baudrate: The baudrate of this ModbusRtuSettings.
    :type: int
    """
    allowed_values = ["1200", "2400", "4800", "9600", "19200"]
    if baudrate not in allowed_values:
        raise ValueError(
            "Invalid value for `baudrate` ({0}), must be one of {1}"
            .format(baudrate, allowed_values)
        )

    self._baudrate = baudrate
```

##### Expected code
```python
    ...
    allowed_values = [1200, 2400, 4800, 9600, 19200]
    if baudrate not in allowed_values:
    ...
```

##### Command line used for generation

So far used/generated via [swagger-editor](http://editor.swagger.io/) ([generator](http://generator.swagger.io/))

**Comment by scottrw93** [comment]
@wing328 Looking at the code [here](https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/python/model.mustache#L63) all enum values are treated as strings. How should maps be handled? Should it just check the values are a subset of the enumerator values?

**Comment by wing328** [comment]
@scottrw93 yes, for Python enum, it needs some work to make it support integer enum similar to what we've done for Java, PHP, etc.

For containers (map, list), it should check to ensure the values are a subset of the enumerator values.

**Comment by wing328** [comment]
@zjarkov 2.3.0 branch has better support for enum in Python API client. Please give that a try and see if you still encounter the same issue.

**Comment by moleksyuk** [comment]
Hi @wing328 

Still have issue with integer enums in Python.

My example:
```yaml
    properties:
      simCard:
        type: "integer"
        format: "int32"
        enum: [1, 2]
        default: 1
        required: false
```
After code generation I'm getting a valid object of enum:
![image](https://user-images.githubusercontent.com/793472/43390397-c7df639e-93f6-11e8-986e-00ae2f27e92e.png)

But a constructor of object with this enum is generated with invalid value:
![image](https://user-images.githubusercontent.com/793472/43390438-dd134442-93f6-11e8-9e5d-7dc6729dd12e.png)

**Comment by wing328** [comment]
@moleksyuk Thanks for tagging me but I'm no longer involved in this project. I hope others will be able to help you out. Good luck.
